aspnetcore/razor-pages/index.md

```diff
@@ -41,35 +43,608 @@ This document provides an introduction to Razor Pages. It's not a step by step t
+
+`@namespace` is explained later in the tutorial. The `@addTagHelper` directive brings in the [built-in Tag Helpers](xref:mvc/views/tag-helpers/builtin-th/Index) to all the pages in the *Pages* folder.
+
+<a name="namespace"></a>
+
+The `@namespace` directive set on a page:
+
+[!code-cshtml[](index/sample/RazorPagesIntro/Pages/Customers/Namespace2.cshtml?highlight=2)]
+
+The `@namespace` directive sets the namespace for the page. The `@model` directive doesn't need to include the namespace.
+
+When the `@namespace` directive is contained in *_ViewImports.cshtml*, the specified namespace supplies the prefix for the generated namespace in the Page that imports the `@namespace` directive. The rest of the generated namespace (the suffix portion) is the dot-separated relative path between the folder containing *_ViewImports.cshtml* and the folder containing the page.
+
+For example, the `PageModel` class *Pages/Customers/Edit.cshtml.cs* explicitly sets the namespace:
+
+[!code-cs[](index/sample/RazorPagesContacts2/Pages/Customers/Edit.cshtml.cs?name=snippet_namespace)]
+
+The *Pages/_ViewImports.cshtml* file sets the following namespace:
+
+[!code-cshtml[](index/sample/RazorPagesContacts2/Pages/_ViewImports.cshtml?highlight=1)]
+
+The generated namespace for the *Pages/Customers/Edit.cshtml* Razor Page is the same as the `PageModel` class.
+
+`@namespace` *also works with conventional Razor views.*
+
+Consider the *Pages/Create.cshtml* view file:
+[!code-cshtml[](index/3.0sample/RazorPagesContacts/Pages/Customers/Create3.cshtml?highlight=2-3)]
+
+The updated *Pages/Create.cshtml* view file with *_ViewImports.cshtml* and the preceding layout file:
+
+[!code-cshtml[](index/3.0sample/RazorPagesContacts/Pages/Customers/Create4.cshtml?highlight=2)]
+
+In the preceding code, the *_ViewImports.cshtml* imported the namespace and Tag Helpers. The layout file imported the JavaScript files.
+
+The [Razor Pages starter project](#rpvs17) contains the *Pages/_ValidationScriptsPartial.cshtml*, which hooks up client-side validation.
+
+For more information on partial views, see <xref:mvc/views/partial>.
+
+<a name="url_gen"></a>
+
+## URL generation for Pages
+
+The `Create` page, shown previously, uses `RedirectToPage`:
 
-::: moniker range="= aspnetcore-2.0"
+[!code-cs[](index/3.0sample/RazorPagesContacts/Pages/Customers/Create.cshtml.cs?name=snippet_PageModel&highlight=15-16)]
 
-Run `dotnet new razor` from the command line.
+The app has the following file/folder structure:
+
+* */Pages*
+
+  * *Index.cshtml*
+  * *Privacy.cshtml*
+  * */Customers*
+
+    * *Create.cshtml*
+    * *Edit.cshtml*
+    * *Index.cshtml*
+
+The *Pages/Customers/Create.cshtml* and *Pages/Customers/Edit.cshtml* pages redirect to *Pages/Customers/Index.cshtml* after success. The string `./Index` is part of the URI to access the preceding page. The string `./Index` can be used to generate URIs to the *Pages/Customers/Index.cshtml* page. For example:
```

Comment:
> The string `./Index` is part of the URI to access the preceding page.

I  think it's really really really important that we don't refer to the strings passed into link generation **as URIs**. This has been a point of confusion for users and I want to make sure we're not giving people the wrong concept.

The correct terminology (page name) is introduced a few paragraphs down and has a good description.

Reply:
@rynowak can you suggest better wording?

Reply:
@scottaddie please suggest better wording.